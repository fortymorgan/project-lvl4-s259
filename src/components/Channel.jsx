import React from 'react';
import { Tooltip, OverlayTrigger } from 'react-bootstrap';
import cn from 'classnames';

const Channel = (props) => {
  const {
    active,
    name,
    handleSwitch,
    handleRemove,
    handleRename,
    removable,
  } = props;

  const generateTooltip = text => (
    <Tooltip id="rename-tooltip">
      {text}
    </Tooltip>
  );

  const removeButton = !removable ? null :
    <OverlayTrigger placement="top" overlay={generateTooltip('Remove')}>
      <span className="delete ml-2" onClick={handleRemove} />
    </OverlayTrigger>;

  const options = (
    <div className="options d-flex">
      <OverlayTrigger placement="top" overlay={generateTooltip('Rename')}>
        <span className="rename" onClick={handleRename} />
      </OverlayTrigger>
      {removeButton}
    </div>
  );

  const className = cn({
    'list-group-item': true,
    'd-flex': true,
    'justify-content-between': true,
    'pr-3': true,
    channel: true,
    active,
  });

  return <li className={className} onClick={handleSwitch}>{name}{options}</li>;
};

export default Channel;
