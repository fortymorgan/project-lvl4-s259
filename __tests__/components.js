import React from 'react';
import renderer from 'react-test-renderer';
import Channel from '../src/components/Channel';

describe('channel component', () => {
  const createChannel = (active, removable) => (
    <Channel
      active={active}
      name="general"
      removable={removable}
    />
  );

  it('non-active non-removable', () => {
    const tree = renderer
      .create(createChannel(false, false))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('active non-removable', () => {
    const tree = renderer
      .create(createChannel(true, false))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('non-active removable', () => {
    const tree = renderer
      .create(createChannel(false, true))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('active removable', () => {
    const tree = renderer
      .create(createChannel(true, true))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
