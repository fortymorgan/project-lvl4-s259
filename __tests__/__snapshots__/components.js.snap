// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`channel component active non-removable 1`] = `
<li
  className="list-group-item d-flex justify-content-between pr-3 channel active"
  onClick={undefined}
>
  general
  <div
    className="options d-flex"
  >
    <span
      className="rename"
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
  </div>
</li>
`;

exports[`channel component active removable 1`] = `
<li
  className="list-group-item d-flex justify-content-between pr-3 channel active"
  onClick={undefined}
>
  general
  <div
    className="options d-flex"
  >
    <span
      className="rename"
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <span
      className="delete ml-2"
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
  </div>
</li>
`;

exports[`channel component non-active non-removable 1`] = `
<li
  className="list-group-item d-flex justify-content-between pr-3 channel"
  onClick={undefined}
>
  general
  <div
    className="options d-flex"
  >
    <span
      className="rename"
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
  </div>
</li>
`;

exports[`channel component non-active removable 1`] = `
<li
  className="list-group-item d-flex justify-content-between pr-3 channel"
  onClick={undefined}
>
  general
  <div
    className="options d-flex"
  >
    <span
      className="rename"
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
    <span
      className="delete ml-2"
      onBlur={[Function]}
      onClick={null}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    />
  </div>
</li>
`;
